pyproject.toml

```diff
@@ -1,3 +1,104 @@
+# https://beta.ruff.rs/docs/configuration
+[tool.ruff]
+target-version = "py39"
+line-length = 120
+select = ["ALL"]
+# https://beta.ruff.rs/docs/rules
+ignore = [
+  ###
+  # Not needed or wanted
+  ###
+  "D1",     # pydocstyle Missing doctring
+  "D401",   # pydocstyle: non-imperative-mood
+  "EM",     # flake8-errmsg
+  "FBT",    # flake8-boolean-trap
+  "INP",    # flake8-no-pep420
+  "ISC003", # flake8-implicit-str-concat: explicit-string-concatenation
+  # Short messages are still considered "long" messages
+  "TRY003", # tryceratops : raise-vanilla-args
+  # Don't remove commented code, also too inconsistant
+  "ERA001", # eradicate: commented-out-code
+  # contextlib.suppress is roughly 3x slower than try/except
+  "SIM105", # flake8-simplify: use-contextlib-suppress
+  # Checked by type-checker (pyright)
+  "ANN", # flake-annotations
+  "TCH", # flake8-type-checking
+  # Already shown by Pylance, checked by pyright, and can be caused by overloads.
+  "ARG002", # Unused method argument
+  # We want D213: multi-line-summary-second-line and D211: no-blank-line-before-class
+  "D203", # pydocstyle: one-blank-line-before-class
+  "D212", # pydocstyle: multi-line-summary-first-line
+
+  ###
+  # Specific to this project
+  ###
+  # We have some Pascal case module names
+  "N999", # pep8-naming: Invalid module name
+  # Print are used as debug logs
+  "T20",  # flake8-print
+  "D205", # Not all docstrings have a short description + desrciption
+
+  ### FIXME (no warnings in Ruff yet: https://github.com/charliermarsh/ruff/issues/1256):
+  "PTH",
```

Comment:
Using pathlib instead of os is the modern way to go. However fixing this could introduce a bug if not careful. So I'm leaving it for later.